alsaaudio.c

```diff
@@ -845,13 +864,173 @@ alsapcm_htimestamp(alsapcm_t *self, PyObject *args)
 }
 
 
-PyDoc_STRVAR(pcm_htimestamp_doc,
+PyDoc_STRVAR(htimestamp_doc,
 "htimestamp() -> tuple\n\
 \n\
 Returns a tuple containing the seconds since epoch in the first element \n\
 , nanoseconds in the second element, and number of frames available in \n\
  the buffer at the time of the time stamp. \n");
 
+
+static PyObject *
+alsapcm_set_tstamp_mode(alsapcm_t *self, PyObject *args)
+{
+	snd_pcm_tstamp_t mode = SND_PCM_TSTAMP_ENABLE;
+	int err;
+
+	if (!PyArg_ParseTuple(args,"|i:set_tstamp_mode", &mode))
+		return NULL;
+				 
+	if (!self->handle)
+	{
+		PyErr_SetString(ALSAAudioError, "PCM device is closed");
+		return NULL;
+	}
+
+	snd_pcm_sw_params_t* swParams;
+	snd_pcm_sw_params_alloca( &swParams);
+
+	snd_pcm_sw_params_current(self->handle, swParams);
+
+	snd_pcm_sw_params_set_tstamp_mode(self->handle, swParams, mode);
+
+	err = snd_pcm_sw_params(self->handle, swParams);
+
+	if (err < 0) {
+		PyErr_SetString(PyExc_RuntimeError, "Unable to set pcm tstamp mode!");
+		return NULL;
+	}
+
+	return Py_None;
```

Comment:
You need a`Py_INCREF(Py_None);` here

Reply:
I see I still need to learn the referencing logic. I'll be teaching the coming days. I'll make the changes next week. But it looks all very doable.

Reply:
Done!